dashboard/app/models/video.rb

```diff
@@ -72,6 +74,12 @@ def self.youtube_base_url
     'https://www.youtube.com'
   end
 
+  def fetch_thumbnail
+    path = dashboard_dir('public', 'c', 'video_thumbnails', "#{key}.jpg")
+    url = "http://img.youtube.com/vi/#{youtube_code}/mqdefault.jpg"
+    IO.copy_stream(open(url), path)
```

Comment:
what's the difference between this and the `IO.write(path, open(url).read)` approach used in fetch_external_resources?

Reply:
Streams the result directly to disk (buffering as necessary) instead of reading the whole thing into memory.  Probably doesn't matter for such a small image 😄.